.github/workflows/ci.yml

```diff
@@ -0,0 +1,15 @@
+on: [push, pull_request]
+
+name: CI
+
+jobs:
+  build_and_test:
+    name: Rust project
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+      - run: cargo build
+      - run: cargo test
```

Comment:
Let's do it as two separate jobs to run them in parallel.

Reply:
I would also add one more job with `cargo clippy -- -D warnings` to check formatting

Reply:
done